engines/dotnet/Main.cs

```diff
@@ -414,44 +394,31 @@ static List<Sample> ModelRegexRedux(Config config)
     // The 'count' function is not part of the measurement.
     static List<Sample> RunAndCount<T>(
         Config config,
-        Count<T> count,
-        Bench<T> bench
+        Func<T, int> count,
+        Func<T> bench
     )
     {
         Stopwatch warmupTimer = Stopwatch.StartNew();
         for (int i = 0; i < config.maxWarmupIters; i++) {
             var result = bench();
             count(result);
-            if (ElapsedNanos(warmupTimer) >= config.maxWarmupTime) {
+            if (warmupTimer.Elapsed.TotalNanoseconds >= config.maxWarmupTime) {
                 break;
             }
         }
 
-        List<Sample> samples = new List<Sample>();
+        List<Sample> samples = new();
         Stopwatch runTimer = Stopwatch.StartNew();
         for (int i = 0; i < config.maxIters; i++) {
             Stopwatch benchTimer = Stopwatch.StartNew();
             var result = bench();
-            var elapsed = ElapsedNanos(benchTimer);
+            var elapsed = benchTimer.Elapsed.TotalNanoseconds;
             var n = count(result);
-            samples.Add(new Sample(elapsed, n));
-            if (ElapsedNanos(runTimer) >= config.maxTime) {
+            samples.Add(new Sample((long)elapsed, n));
+            if (runTimer.Elapsed.TotalNanoseconds >= config.maxTime) {
                 break;
             }
         }
         return samples;
     }
-
-    // Return the elapsed time on the given stop watch in terms of
-    // nano-seconds.
-    //
-    // Note that .NET doesn't guarantee that the stop-watch uses nanosecond
-    // resolution. So this might only be, for example, capable of returning
-    // nanoseconds in intervals of 100. But the units returned are indeed
-    // always nanoseconds.
```

Comment:
Did this guarantee change? That is, does `Elapsed.TotalNanoseconds` guarantee nanosecond resolution? If not, it might make sense to keep this comment somewhere as a note.

Reply:
I'll put the comment back. It's still dependent on whatever resolution is supported by the underlying operating system functionality being employed; Stopwatch uses whatever the most precise clock is that's available, but on some OSes there's nothing as fine grained as 1ns.

Reply:
That said, I don't see the same caveat on other implementations here, and the language can't do better than what the OS/hardware provides. The .NET implementation does use the best clock available.

Reply:
Hmmm all right, you can drop it then. I probably came to the same conclusion. .NET was one of the first runner programs I implemented (outside of Rust).